docs/reference/search/suggesters/context-suggest.asciidoc

```diff
@@ -1,237 +1,269 @@
+    This is mandatory.
+
+`boost`::
+    The factor by which the score of the suggestion
+    should be boosted, the score is computed by
+    multiplying the boost with the suggestion weight,
+    defaults to `1`
+
+`prefix`::
+    Whether the context value should be treated as a
+    prefix or not. For example, if set to `true`,
+    you can filter context of 'type1', 'type2' and
+    so on, by specifying a context prefix of 'type'.
+    Defaults to `false`
 
 [[suggester-context-geo]]
 [float]
 ==== Geo location Context
-A `geo` context allows you to limit results to those that lie within a certain distance
-of a specified geolocation. At index time, a lat/long geo point is converted into a
-geohash of a certain precision, which provides the context.
+
+A `geo` context allows you to associate one or more geo point or geohash with suggestions
+at index time. At query time, suggestions can be filtered and boosted if they are within
+a certain distance of a specified geo location.
 
 [float]
-===== Geo location Mapping
-The mapping for a geo context accepts four settings, only of which `precision` is required:
+===== Geo Mapping
+
+In addition to `path` setting, `geo` context mapping accepts the following settings:
 
 [horizontal]
-`precision`::  This defines the precision of the geohash and can be specified as `5m`, `10km`,
-               or as a raw geohash precision: `1`..`12`. It's also possible to setup multiple
-               precisions by defining a list of precisions: `["5m", "10km"]`
-`neighbors`::  Geohashes are rectangles, so a geolocation, which in reality is only 1 metre
-               away from the specified point, may fall into the neighbouring rectangle. Set
-               `neighbours` to `true` to include the neighbouring geohashes in the context.
-               (default is *on*)
-`path`::       Optionally specify a field to use to look up the geopoint.
-`default`::    The geopoint to use if no geopoint has been specified.
-
-Since all locations of this mapping are translated into geohashes, each location matches
-a geohash cell. So some results that lie within the specified range but not in the same
-cell as the query location will not match. To avoid this the `neighbors` option allows a
-matching of cells that join the bordering regions of the documents location. This option
-is turned on by default.
-If a document or a query doesn't define a location a value to use instead can defined by
-the `default` option. The value of this option supports all the ways a `geo_point` can be
-defined. The `path` refers to another field within the document to retrieve the
-location. If this field contains multiple values, the document will be linked to all these
-locations.
+`precision`::
+    This defines the precision of the geohash to be indexed and can be specified
+    as a distance value (`5m`, `10km` etc.), or as a raw geohash precision (`1`..`12`).
+    Defaults to a raw geohash precision value of `12`.
```

Comment:
Is this the right default?  Won't it make for very big FSTs? It amounts to centimetre level, which seems way over the top.  I'd think km level (6) would be a better default